In this exercise, you will practice basic `dplyr` (the relevant package within `tidyverse`) operations on a dataset of gene expression levels across human tissues (from the GTEx Project). The dataset (`dt`) has one row per gene, with columns:

-   `Name`: the Ensembl gene ID\
-   `Description`: the gene symbol
-   One column per tissue containing the median expression levels (in TPM units)

------------------------------------------------------------------------

### Start by loading the data into R:

Note the arguments, which specify that the first two lines of the file will be skipped, that the file contains a header line that should be used to form column names, and that the columns are tab (`\t`) separated.

```{r}
dt <- read_delim("GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_median_tpm.gct", skip = 2, col_names = TRUE, delim = "\t")
```

### Q1. How many genes are in the dataset, and how many tissues are represented?

```{r}
# code and answer here
dim(dt)
#There is 56200 genes and 56-2= 54 tissues represented

```

------------------------------------------------------------------------

### Q2. How many genes are expressed above a given threshold in each tissue?

A common threshold is 1 TPM to define whether a gene is "expressed."\
Compute, for each of the three tissues, `Liver`, `Testis`, and `Whole Blood`, how many genes are expressed above this threshold.\
Then identify which tissue has the **most** and which has the **fewest** expressed genes. See this paper for one possible explanation: Xia et al. 2021: [PMC7891839](https://pmc.ncbi.nlm.nih.gov/articles/PMC7891839/).

Do the results change if you set the threshold higher, say at 10 TPM?

```{r}
# Code here
dt %>%
  filter(Liver >1) %>% 
  summarise(Liverexpressed= n())

# Amount of genes that are expressed above 1 TPM for liver is 13499

# LiverGreaterThanOne <- dt %>% filter(Liver > 1)
# print(nrow(LiverGreaterThanOne))
# 13,499

dt %>%
  filter(Testis>1) %>%
  summarise(Testisexpressed= n())
# Amount of genes that are expressed above 1 TPM for Testis is 25064

dt %>%
  filter(`Whole Blood`>1) %>%
  summarise(Whole_Bloodexpressed= n())
# Amount of genes that are expressed above 1 TPM for Whole Blood is 11616

# Whole Blood has the fewest expressed genes while Testis has the most expressed genes

dt %>%
  filter(Liver >10) %>% 
  summarise(Liverexpressed= n())
# Amount of genes that are expressed above 10 TPM for Liver is 5132

dt %>%
  filter(Testis>10) %>%
  summarise(Testisexpressed= n())
# Amount of genes that are expressed above 10 TPM for Testis is 11471


dt %>%
  filter(`Whole Blood`>10) %>%
  summarise(Whole_Bloodexpressed= n())
# Amount of genes that are expressed above 10 TPM for Whole Blood is 4482

# Interpretation here as a comment

```

------------------------------------------------------------------------

### Q3. What are the 20 most highly expressed genes in the liver?

What do you notice about many of the genes in this list? Can you propose a hypothesis to explain your observation?

```{r}
# Code here
dt %>%
  select(Name, Description, Liver) %>%
  arrange(desc(Liver)) %>%
  slice_head(n=20)

#Many of the genes in this list are "MT" in descriptions, meaning they are mitochondria genes. This could mean that the liver uses many mitochondria genes. 

# Interpretation here as a comment
# ENSG00000198886.2	MT-ND4	58775.30		
# ENSG00000198899.2	MT-ATP6	51821.40		
# ENSG00000198712.1	MT-CO2	49563.60		
# ENSG00000198938.2	MT-CO3	44570.80		
# ENSG00000198804.2	MT-CO1	40518.90		
# ENSG00000198727.2	MT-CYB	31641.50		
# ENSG00000228253.1	MT-ATP8	29932.50		
# ENSG00000212907.2	MT-ND4L	26085.50		
# ENSG00000163631.16	ALB	25201.30		
# ENSG00000198763.3	MT-ND2	22892.20
# ENSG00000198888.2	MT-ND1	20800.20		
# ENSG00000198840.2	MT-ND3	19264.40		
# ENSG00000210082.2	MT-RNR2	18805.50		
# ENSG00000257017.8	HP	12259.20		
# ENSG00000173432.10	SAA1	8842.00		
# ENSG00000229314.5	ORM1	8805.70		
# ENSG00000087086.14	FTL	7717.98		
# ENSG00000110245.11	APOC3	7599.74		
# ENSG00000171564.11	FGB	7351.90		
# ENSG00000198786.2	MT-ND5	6071.64	

```

------------------------------------------------------------------------

### Q4. Compare the mean expression levels of mitochondrial versus nuclear-encoded genes in the Liver.

Genes encoded by the mitochondrial genome have the gene symbol prefix `MT-`. The function `str_starts()` from the `stringr` package in `tidyverse` tests whether a string (or a vector of strings) start with a given pattern, returning TRUE/FALSE (or a boolean vector) as output.

Use `str_starts()` to identify which genes are mitochondrially encoded and store this information in a new Boolean column named `is_mitochondrial`. Next, compute the mean and median expression of mitochondrial genes in the liver, as well as the mean and median expression of nuclear-encoded genes in the liver and compare them. Do mitochondrial genes exhibit higher or lower mean and median expression than nuclear genes? What does the difference between the mean and median tell you about the distribution of gene expression.

```{r}
# Code here
library(tidyverse)

dt <- dt %>%
  mutate(is_mitochondrial = str_starts(Description, "MT-")) %>%
  select(Name, Description, Liver, is_mitochondrial) %>%
  filter(is_mitochondrial == TRUE)


# dt %>%
#   left_join(is_mitochondrial, by = "Description") %>%


dt %>%
  group_by(is_mitochondrial) %>%
  summarise(mean_expression = mean(Liver)) 

dt %>%
  group_by(is_mitochondrial) %>%
  summarise(median_expresssion = median(Liver))

dt <- dt %>%
  mutate(is_Nuclear = str_starts(Description, "MT-")) %>%
  select(Name, Description, Liver, is_Nuclear) %>%
  filter(is_Nuclear == FALSE)
dt

dt %>%
  group_by(is_Nuclear) %>%
  summarise(mean_expression = mean(Liver)) 

dt %>%
  group_by(is_Nuclear) %>%
  summarise(median_expresssion = median(Liver))

             
            
  


# Interpretation here as a comment
# Mitochondrial mean expression= 11592.72
# Mitochondrial median expression= 3.40258
# Nuclear mean expression = 3.373839
# Nuclear median expression = 0

# Mitochondrial genes exhibit higher mean and median expression than nuclear genes. The difference between the mean and median tells me that within the liver there is more distribution of Mitochondrial gene expression.

```

------------------------------------------------------------------------

### Q5. Are liverâ€™s most highly expressed genes also highly expressed in another tissue?

Take the 20 most highly exprssed genes in **Liver** and check how they are expressed in the **Heart - Left Ventricle**. Identify:\
1. one gene that is highly expressed in both tissues\
2. one gene that is highly expressed in Liver but very low in Heart

Look up basic information about these genes on the internet. Do their expression patterns make sense given their known functions?

```{r}

 # Code here
x<-dt %>%
  select(Name, Description, Liver) %>%
  arrange(desc(Liver)) %>%
  slice_head(n=20) 
# ENSG00000198886.2	MT-ND4	58775.30		
# ENSG00000198899.2	MT-ATP6	51821.40		
# ENSG00000198712.1	MT-CO2	49563.60		
# ENSG00000198938.2	MT-CO3	44570.80		
# ENSG00000198804.2	MT-CO1	40518.90		
# ENSG00000198727.2	MT-CYB	31641.50		
# ENSG00000228253.1	MT-ATP8	29932.50		
# ENSG00000212907.2	MT-ND4L	26085.50		
# ENSG00000163631.16	ALB	25201.30		
# ENSG00000198763.3	MT-ND2	22892.20	
# ENSG00000198888.2	MT-ND1	20800.20		
# ENSG00000198840.2	MT-ND3	19264.40		
# ENSG00000210082.2	MT-RNR2	18805.50		
# ENSG00000257017.8	HP	12259.20		
# ENSG00000173432.10	SAA1	8842.00		
# ENSG00000229314.5	ORM1	8805.70		
# ENSG00000087086.14	FTL	7717.98		
# ENSG00000110245.11	APOC3	7599.74		
# ENSG00000171564.11	FGB	7351.90		
# ENSG00000198786.2	MT-ND5	6071.64	
 

dt %>%
  select(Description) %>%
  arrange(desc(Liver)) %>%
  slice_head(n=20)



x <- x %>% select(Description) %>% c()

xy <- x$Description

dt %>%
  select(Description,`Heart - Left Ventricle`) %>%
  filter(Description %in% xy) %>%
  arrange(desc(xy))





# Interpretation here as a comment


# one gene that is highly expressed in both tissues: MT-CO3
# one gene that is highly expressed in Liver but very low in Heart: APOC3
# The MT-CO3 gene is needed for cellular respiration, a process required in both the heart and liver. In contrast, the APOC3 gene regulates triglyceride metabolism in the bloodstream, a function primarily carried out by the liver, not the heart.
```

------------------------------------------------------------------------

### Q6. Comparing gene expression between tissues

Another way to compare tissues is to ask how similar are their expression patterns overall, across all genes?

R has a built-in function called `cor.test()` that measures the correlation between two vectors of numbers. Remember that you can extract the contents of a given column of a tibble as a vector using the syntax `tibble_name$column_name`. If a column name contains spaces, enclose the column name in backticks such as `` tibble_name$`column name` ``.

Use `cor.test()` with the argument `method = "kendall"` to compute: - the Spearman correlation between `Liver` and `Heart - Left Ventricle` - the Spearman correlation between `Liver` and `Brian - Cortex`

Which pair of tissues is more strongly correlated? Does this result make sense biologically?

```{r}

Liver_x<-dt %>%
  select(Name, Description, Liver) %>%
  slice_head(n= 56200) 

Heart_x<-dt %>%
  select(Name, Description,`Heart - Left Ventricle` ) %>%
  slice_head(n= 56200) 

Brain_x<-dt %>%
  select(Name, Description, `Brain - Cortex`) %>%
  slice_head(n= 56200) 
  

Liver_x<- Liver_x %>% select(Liver) %>% c()

Liver_x<- Liver_x$Liver

Heart_x<- Heart_x %>% select(`Heart - Left Ventricle`) %>% c()

Heart_x<- Heart_x$`Heart - Left Ventricle`

Brain_x<- Brain_x %>% select(`Brain - Cortex`) %>% c()

Brain_x<- Brain_x$`Brain - Cortex`

cor.test(Liver_x,Heart_x, method="kendall")

cor.test(Liver_x,Brain_x, method="kendall")












# Liver vs Heart answer

# Kendall's rank correlation tau

# data:  Liver_x and Heart_x
# z = 241.7, p-value < 2.2e-16
# alternative hypothesis: true tau is not equal to 0
# sample estimates:
#       tau 
# 0.8053218 


# Liver vs Brain answer

# Kendall's rank correlation tau
# 
# data:  Liver_x and Brain_x
# z = 223.7, p-value < 2.2e-16
# alternative hypothesis: true tau is not equal to 0
# sample estimates:
#       tau 
# 0.7342544 

# interpretation

# Liver and Heart - Left Ventricle is more strongly correlated. This result makes biological sense, as the liver and heart share similar functions, such as their involvement in the circulatory system and blood circulation.
```

------------------------------------------------------------------------

### Optional: further exploration

What other patterns are present in the data? Use tidyverse functions to explore and summarize the data in different ways and see what you can learn.

```{r}
# answer here
```
